packaging/windows/install_script.nsis

```diff
@@ -0,0 +1,237 @@
+Var RemoteWriteText
+
+Var RemoteWriteUserLabel
+Var RemoteWriteUserText
+
+Var RemoteWritePasswordLabel
+Var RemoteWritePasswordText
+
+Var EnableExporterLabel
+Var EnableExporterCheck
+
+# These are strings that hold the values
+
+Var RemoteUrl
+Var RemoteUser
+Var RemotePassword
+Var EnableExporterValue
+ 
+page directory
+Page custom remoteWriteInfo remoteWriteInfoLeave
+Page custom enableWindowsExporter enableWindowsExporterLeave
+Page instfiles
+
+!macro VerifyUserIsAdmin
+UserInfo::GetAccountType
+pop $0
+${If} $0 != "admin" ;Require admin rights on NT4+
+        messageBox mb_iconstop "Administrator rights required!"
+        setErrorLevel 740 ;ERROR_ELEVATION_REQUIRED
+        quit
+${EndIf}
+!macroend
+
+Function remoteWriteInfo
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+    ${NSD_CreateLabel} 0 0 100% 12u "Enter URL for remote write"
+    Pop $RemoteWriteLabel
+
+    ${NSD_CreateText} 0 13u 100% 12u ""
+    Pop $RemoteWriteText
+
+    ${NSD_CreateLabel} 0 26u 100% 12u "Enter User for remote write"
+    Pop $RemoteWriteUserLabel
+
+    ${NSD_CreateText} 0 39u 100% 12u ""
+    Pop $RemoteWriteUserText
+
+    ${NSD_CreateLabel} 0 52u 100% 12u "Enter Password for remote write"
+    Pop $RemoteWritePasswordLabel
+
+    ${NSD_CreateText} 0 65u 100% 12u ""
+    Pop $RemoteWritePasswordText
+
+    nsDialogs::Show
+FunctionEnd
+
+
+Function remoteWriteInfoLeave
+    ${NSD_GetText}  $RemoteWriteText  $RemoteUrl
+    ${NSD_GetText}  $RemoteWriteUserText  $RemoteUser
+    ${NSD_GetText}  $RemoteWritePasswordText  $RemotePassword
+FunctionEnd
+ 
+Function enableWindowsExporter
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+
+    ${NSD_CreateLabel} 0 0 100% 12u "Add and Enable Windows Exporter"
+    Pop $EnableExporterLabel
+
+    ${NSD_CreateCheckBox} 0 13u 100% 12u ""
+    Pop $EnableExporterCheck
+
+    nsDialogs::Show
+FunctionEnd
+
+Function enableWindowsExporterLeave
+    ${NSD_GetState} $EnableExporterCheck $EnableExporterValue
+FunctionEnd
+ 
+ 
+function .onInit
+    setShellVarContext all
+    !insertmacro VerifyUserIsAdmin
+functionEnd
+ 
+section "install"
+    # Files for the install directory - to build the installer, these should be in the same directory as the install script (this file)
+    setOutPath $INSTDIR
+    # Files added here should be removed by the uninstaller (see section "uninstall")
+    file "agent-windows-amd64.exe"
+    file "logo.ico"
+    file "nssm.exe"
+    file "windows_exporter-0.15.0-amd64.exe"
+    # Add any other files for the install directory (license files, app data, etc) here
+ 
+    # Uninstaller - See function un.onInit and section "uninstall" for configuration
+    writeUninstaller "$INSTDIR\uninstall.exe"
+ 
+    # Start Menu
+    createDirectory "$SMPROGRAMS\"
+    createShortCut "$SMPROGRAMS\${APPNAME}.lnk" "$INSTDIR\agent-windows-amd64.exe" "" "$INSTDIR\logo.ico"
```

Comment:
I'm not sure I'm following why we create a shortcut here since `agent-windows-amd64.exe` is a command line application. Should e remove the shortcut?

Reply:
Probably doesn't make any sense since they would need to pass config lines.